net/pubsub.go

```diff
@@ -140,7 +140,9 @@ func (s *PubSub) watch(ctx context.Context, id thread.ID, topic *topic) {
 // subscribe to a topic for thread updates.
 func (s *PubSub) subscribe(ctx context.Context, id thread.ID, topic *topic) {
 	var err error
+	s.Lock()
 	topic.s, err = topic.t.Subscribe()
+	s.Unlock()
```

Comment:
I'd like @sanderpick  opinion about this one.

First and foremost, this lock is necessary since `Remove()` reads `topic.s` too. `subscribe()` is called from `Add()`, so on `Add()` and `Remove()` concurrent calls, there is a data-race on `topic.s`. So, with this lock that gets fixed.

Now, as `Add` comments says `Add()` can be called multiple times. And and `Add` call, will lead to a `subscribe()` call, which here looks like it's assigning `topic.s` which is an instance-wide attribute. Is that OK? Seems that an `Add` call will override a previous `Add` call. Mentioning this to be sure that was intentional.